vulcanai2/models/basenetwork.py

```diff
@@ -66,7 +66,14 @@ def __init__(self, name, dimensions, config, save_path=None, input_network=None,
 
         self._save_path = save_path
 
-        self._input_network = input_network
+        self._input_networks = input_networks
+        if isinstance(input_networks, list) and len(input_networks)==1:
+            self._input_network = input_networks[0]
```

Comment:
using both `self._input_networks` and `self._input_network` is confusing. you should just stick to using only `self._input_networks` so we have one point of entry for getting access to all incoming networks (even if it's just one)

Reply:
agreed

Reply:
actually, wait. that's what he did though.